src/files/files.go

```diff
@@ -0,0 +1,341 @@
+		for scanner.Scan() {
+			currentLineText := scanner.Text()
+			outputLineText := []byte(fmt.Sprintf("%s\n", currentLineText))
+
+			if f.containsVariableNameInText(currentLineText, currentFile.VariableName, ":=") {
+				outputLineText = []byte(fmt.Sprintf("%s := \"%s\"\n", currentFile.VariableName, newVersion))
+				variableNameFound = true
+			}
+
+			if f.containsVariableNameInText(currentLineText, currentFile.VariableName, "=") {
+				outputLineText = []byte(fmt.Sprintf("%s = \"%s\"\n", currentFile.VariableName, newVersion))
+				variableNameFound = true
+			}
+
+			outputData = append(outputData, outputLineText...)
+		}
+
+		if !variableNameFound {
+			return fmt.Errorf("variable name `%s` not found on file `%s`", currentFile.VariableName, currentFile.Path)
+		}
+
+		if err := scanner.Err(); err != nil {
+			return err
+		}
+
+		destinationPath := f.setDefaultPath(currentFile.DestinationPath, currentFile.Path)
+		if err = ioutil.WriteFile(destinationPath, outputData, 0666); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (f *FileVersion) validateChangesInfo(changelog ChangesInfo) error {
+
+	if changelog.AuthorName == "" {
+		return errors.New("author name cannot be empty")
+	}
+
+	if changelog.AuthorEmail == "" || !strings.Contains(changelog.AuthorEmail, "@") {
+		return errors.New("bad author email entry")
+	}
+
+	if changelog.ChangeType == "" {
+		return errors.New("change type cannot be empty")
+	}
+
+	if len(changelog.Hash) < 7 {
+		return errors.New("hash string must have at least 7 characters")
+	}
+
+	if changelog.Message == "" {
+		return errors.New("message cannot be empty")
+	}
+
+	if changelog.CurrentVersion == "" {
+		return errors.New("current version cannot be empty")
+	}
+
+	if changelog.NewVersion == "" {
+		return errors.New("new version cannot be empty")
+	}
+
+	return nil
+}
+
+func (f *FileVersion) abbreviateHash(hash string) string {
+	return hash[:7]
+}
+
+func (f *FileVersion) getCommitUrl(changelog ChangesInfo) string {
+	return fmt.Sprintf("[%s](https://%s/%s/%s/commit/%s)", f.abbreviateHash(changelog.Hash), f.versionConrolHost, f.groupName, f.projectName, changelog.Hash)
+}
+
+func (f *FileVersion) prettifyEmail(email string) string {
+	splitedEmail := strings.Split(email, "@")
+	return fmt.Sprintf("@%s", splitedEmail[0])
+}
+
+func (f *FileVersion) unmarshalChangesInfo(changes interface{}) (*ChangesInfo, error) {
+	changeLogInfoBytes, err := json.Marshal(changes)
+	if err != nil {
+		return nil, errors.New("error marshalling files to changelog information")
+	}
+
+	var changelog ChangesInfo
+	if err := json.Unmarshal(changeLogInfoBytes, &changelog); err != nil {
+		return nil, errors.New("error unmarshalling changelog information")
+	}
+
+	return &changelog, nil
+}
+
+// UpgradeChangelog aims to append the new release version with the commit information to the CHANGELOG.md file.
+func (f *FileVersion) UpgradeChangeLog(path, destinationPath string, chageLogInfo interface{}) error {
```

Comment:
maybe split this function in sub function... @esequielvirtuoso

Reply:
@LeoCBS 
I've changed it a little bit. Can you verify that once again, please?